### ISSUE TITLE
make usernames of deleted inactive users available for new users

### ISSUE THREAD

**Issue by splacat** [mastodon/mastodon#19634]
### Pitch

Actually new users can't use many usernames because many of them are not available but used during sign up by already delete inactive users.
One of my server's rules is like this, all inactive users will be warned by email after six months of activity and they have 30 more days to sign in to stop the deletion process. Most of them never stop the process so they are deleted but their usernames get "blocked" forever and most of them never posted nothing. 

### Motivation

Could be good to get able to get all usernames of deleted users with zero post available again for a new ones.

**Comment by splacat** [comment]
It's is like "you never used your username?" we will give it to a new user

**Comment by Ry3yr** [comment]
This "blocking of retaking usernames" is actually one of Mastodon's SECURITY FEATURES to orevent impersonation by third parties post account deletion.

Even if 0 posts.

For comparison:
Twitter has /no such feature/ ...

**Comment by supernovae** [comment]
I think it's worth re-visiting this. If someone has never posted, never done a profile pic, never done anything they didn't reserve anything and nothing is protected by keeping them around. We should be able to purge these to free up usernames.

**Comment by JustDalek** [comment]
I would be interested in this as well! I was just looking into a way to release old usernames from any accounts with zero posts or media on my instance.

Would also love some sort of activity requirement and purging if inactive say for more than 1 year or 2 perhaps

**Comment by tiagovercosa** [comment]
I agree with the proposal. **In active instances, usernames end up becoming a limited resource**, and keeping them tied to accounts with no activity harms new users without bringing real benefit to the network.

A balanced approach would be to allow release only of fully inactive accounts (with no posts, followers, or interactions), accompanied by prior notice and a grace period for login. Additionally, keeping the username in quarantine for a period before reusing it would help prevent impersonation and disputes over names. This preserves users’ identity and, at the same time, improves the availability of handles.
